### PR TITLE
JNG-4188 Create method should not be generated for on abstract entities

### PR DESCRIPTION
**Pull request by bencegelei**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4188" title="JNG-4188" target="_blank">TaskJNG-4188</a>  Create method should not be generated for on abstract entities
  </td></table>
  <br />
 

JNG-4188 Create method should not be generated for on abstract entities

### CODE REVIEW

**Review by noherczeg — APPROVED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/QueryTest.java [hunk 1]
```diff
@@ -101,18 +99,6 @@ public void testLimit() {
...
-        myEntityWithOptionalFieldsDao.delete(entity1);
-
-        IllegalArgumentException thrown = assertThrows(
-                IllegalArgumentException.class,
-                () -> myEntityWithOptionalFieldsDao.getById(entity1.get__identifier())
-        );
-
-        assertNotNull(thrown);
-    }
-
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/QueryTest.java:115]: getById returns Optional object if entity is not found therefore it should not throw any exception.

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/SalesModelTest.java [hunk 2]
```diff
@@ -83,15 +84,6 @@ public void test() {
...
 
-        Person createdPerson = personDao.create(Person.builder()
-                .withFirstName("Masik")
-                .withLastName("Test")
-                .build());
-
-        assertEquals(Optional.of("Masik"), createdPerson.getFirstName());
-        assertEquals(Optional.of("Test"), createdPerson.getLastName());
-        assertEquals(Optional.of("Masik Test"), createdPerson.getFullName());
-
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/SalesModelTest.java:94]: Person is abstract, create method is now not generated

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/SalesModelTest.java [hunk 3]
```diff
@@ -195,17 +187,6 @@ public void testNavigationOnStaticCollection() {
...
-    public void testErrorRequiredFieldValidation() {
-        IllegalArgumentException thrown = assertThrows(
-                IllegalArgumentException.class,
-                () -> MyError.builder().build()
-        );
-
-        assertTrue(thrown.getMessage().contains("missing mandatory attribute"));
-        assertTrue(thrown.getMessage().contains("name: code"));
-    }
-
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/SalesModelTest.java:208]: Errors do not have required fields and based on the latest documentation they are not required by default either

File: judo-runtime-core-jsl-spring-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/spring/JudoRuntimeCoreSpringApplicationTests.java [hunk 4]
```diff
@@ -65,14 +65,6 @@ void testDaoFunctions() {
...
 		assertEquals(1, personList.size());
 
-		Person createdPerson = personDao.create(Person.builder()
-				.withFirstName("Masik")
-				.withLastName("Test")
-				.build());
-
-		assertEquals(Optional.of("Masik"), createdPerson.getFirstName());
-		assertEquals(Optional.of("Test"), createdPerson.getLastName());
-
```
Comment (bencegelei) [inline judo-runtime-core-jsl-spring-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/spring/JudoRuntimeCoreSpringApplicationTests.java:75]: Person is abstract, create method is now not generated

**Review by noherczeg — APPROVED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/SalesModelTest.java (hunk 2 above)
Comment (robertcsakany) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/SalesModelTest.java:94]: :)

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**